java/core/src/main/java/com/microsoft/bond/BondType.java

```diff
@@ -0,0 +1,514 @@
+     */
+    protected abstract T deserializeValue(TaggedDeserializationContext context) throws IOException;
+
+    /**
+     * Serializes a struct field of this type into a protocol writer, including field metadata.
+     * This method is not recommended for fields with Java primitive (non-object) types. To serialize fields with
+     * primitive values use the static helper method defined in each singleton class for a primitive type.
+     *
+     * @param context contains the runtime context of the serialization
+     * @param value   the value to serialize
+     * @param field   descriptor of the field
+     * @throws IOException if an I/O error occurred
+     */
+    protected abstract void serializeField(
+            SerializationContext context,
+            T value,
+            StructBondType.StructField<T> field) throws IOException;
+
+    /**
+     * Serializes a struct field of this type with "nothing" as the default value into a protocol writer,
+     * including field metadata.
+     * This method is not recommended for fields with Java primitive (non-object) types. To serialize fields with
+     * primitive values use the static helper method defined in each singleton class for a primitive type.
+     *
+     * @param context contains the runtime context of the serialization
+     * @param value   the value to serialize
+     * @param field   descriptor of the field
+     * @throws IOException if an I/O error occurred
+     */
+    protected final void serializeSomethingField(
+            SerializationContext context,
+            SomethingObject<T> value,
+            StructBondType.StructField<T> field) throws IOException {
+        if (value != null) {
+            serializeField(context, value.getValue(), field);
+        }
+    }
+
+    /**
+     * Deserializes a struct field of this type from a tagged protocol reader, excluding field metadata
+     * which is assumed to be already deserialized earlier and whose value is available in the
+     * {@link TaggedDeserializationContext#readFieldResult} field of the passed context argument.
+     * This method is not recommended for fields with Java primitive (non-object) types. To deserialize fields with
+     * primitive values use the static helper method defined in each singleton class for a primitive type.
+     *
+     * @param context contains the runtime context of the deserialization
+     * @param field   descriptor of the field
+     * @return the deserialized value
+     * @throws IOException if an I/O error occurred
+     */
+    protected abstract T deserializeField(
+            TaggedDeserializationContext context,
+            StructBondType.StructField<T> field) throws IOException;
+
+    /**
+     * Deserializes a struct field of this type with "nothing" as the default value from a tagged protocol reader,
+     * excluding field metadata which is assumed to be already deserialized earlier and whose value is available
```

Comment:
It never returns null as is clear from the code. Its purpose is to deserialize a value and then wrap it by "something". I don't see anything strange here.

---
In reply to: [123094559](https://github.com/Microsoft/bond/pull/502#discussion_r123094559) [](ancestors = 123094559)